include/zos-tls.h

```diff
@@ -37,4 +37,39 @@ template <typename T> class __Z_EXPORT __tlssim {
 };
 
 #endif // __cplusplus
+
+/* 
+These macros provide a workaround for the non-existing thread local storage (TLS) 
```

Comment:
I tend to prefer functions over macros since they're easier to debug. You could also add the inline attribute if you prefer the code to be inlined.